### ISSUE TITLE
Error: Invalid prefix for given prefix length

### ISSUE THREAD

**Issue by jeffmccune** [openinfrastructure/terraform-google-multinic#28]
Using Version 2.0.0, Potentially related to:

```
Dec 23 16:25:59 multinic-us-east1-trxs systemd[1]: Started System Logging Service.
Dec 23 16:25:59 multinic-us-east1-trxs policy-routing[1587]: Error: Invalid prefix for given prefix length.
Dec 23 16:25:59 multinic-us-east1-trxs policy-routing[1587]: Error: Invalid prefix for given prefix length.
Dec 23 16:25:59 multinic-us-east1-trxs policy-routing[1587]: Error: Invalid prefix for given prefix length.
Dec 23 16:25:59 multinic-us-east1-trxs policy-routing[1587]: Error: Invalid prefix for given prefix length.
Dec 23 16:25:59 multinic-us-east1-trxs systemd[1]: Started Configure Policy Routing to behave as a virtual wire.
```

"This is causing the health checks to fail on new instances."

**Comment by jeffmccune** [comment]
Workaround for 2.0.0
===

Pin the image in 2.0.0 as a workaround.

Find a suitable image from the past:

```
gcloud compute images list --project centos-cloud --no-standard-images --show-deprecated
```

Pin the version using the os_image input var:

```diff
diff --git a/examples/compute/main.tf b/examples/compute/main.tf
index cb70663..c854ab0 100644
--- a/examples/compute/main.tf
+++ b/examples/compute/main.tf
@@ -58,6 +58,7 @@ module "multinic" {
   name_prefix = "multinic"
   region      = local.region
   zones       = [ "${local.region}-b", "${local.region}-c" ]
+  os_image    = "centos-cloud/centos-8-v20200811"

   nic0_network = local.nic0_network
   nic0_project = local.project_id
```

Apply the change and the health checks should pass:

<img width="1634" alt="Screen Shot 2020-12-23 at 11 38 58 AM" src="https://user-images.githubusercontent.com/8972/103032009-dbd81a00-4513-11eb-9df5-216286d743c3.png">

<img width="1104" alt="Screen Shot 2020-12-23 at 11 40 08 AM" src="https://user-images.githubusercontent.com/8972/103032018-e2669180-4513-11eb-83cd-fff4d2b63654.png">

<img width="1098" alt="Screen Shot 2020-12-23 at 11 40 50 AM" src="https://user-images.githubusercontent.com/8972/103032029-e8f50900-4513-11eb-9c52-836ca6918466.png">

**Comment by jeffmccune** [comment]
With version 3.2.0

```diff
diff --git a/modules/50_compute/files/startup-multinic.sh b/modules/50_compute/files/startup-multinic.sh
index f2f8cb3..5f7b6c7 100755
--- a/modules/50_compute/files/startup-multinic.sh
+++ b/modules/50_compute/files/startup-multinic.sh
@@ -229,6 +229,7 @@ configure_policy_routing() {
   tmpfile="$(mktemp)"
   cat <<EOF >"$tmpfile"
 #! /bin/bash
+set -x
 # These tables manage default routes based on policy.
 if ! grep -qx '10 viaeth0' /etc/iproute2/rt_tables; then
   echo "10 viaeth0" >> /etc/iproute2/rt_tables
```

```
[jeff@multinic-phk3 ~]$ sudo journalctl -xu policy-routing
-- Logs begin at Thu 2020-12-24 02:35:17 UTC, end at Thu 2020-12-24 02:36:12 UTC. --
Dec 24 02:35:36 multinic-phk3 systemd[1]: Starting Configure Policy Routing to behave as a virtual wire...
-- Subject: Unit policy-routing.service has begun start-up
-- Defined-By: systemd
-- Support: https://access.redhat.com/support
--
-- Unit policy-routing.service has begun starting up.
Dec 24 02:35:36 multinic-phk3 policy-routing[1635]: + grep -qx '10 viaeth0' /etc/iproute2/rt_tables
Dec 24 02:35:36 multinic-phk3 policy-routing[1635]: + echo '10 viaeth0'
Dec 24 02:35:36 multinic-phk3 policy-routing[1635]: + grep -qx '11 viaeth1' /etc/iproute2/rt_tables
Dec 24 02:35:36 multinic-phk3 policy-routing[1635]: + echo '11 viaeth1'
Dec 24 02:35:36 multinic-phk3 policy-routing[1635]: + ip route replace default via 10.33.0.1 dev eth0 proto static table viaeth0
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.33.0.1 dev eth0 proto static scope link table viaeth0
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.37.0.1 dev eth1 proto static scope link table viaeth0
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.33.0.25/20 via 10.33.0.1 dev eth0 proto static table viaeth0
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: Error: Invalid prefix for given prefix length.
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.37.0.25/20 via 10.37.0.1 dev eth1 proto static table viaeth0
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: Error: Invalid prefix for given prefix length.
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace default via 10.37.0.1 dev eth1 proto static table viaeth1
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.33.0.1 dev eth0 proto static scope link table viaeth1
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.37.0.1 dev eth1 proto static scope link table viaeth1
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.33.0.25/20 via 10.33.0.1 dev eth0 proto static table viaeth1
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: Error: Invalid prefix for given prefix length.
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route replace 10.37.0.25/20 via 10.37.0.1 dev eth1 proto static table viaeth1
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: Error: Invalid prefix for given prefix length.
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip rule add priority 1000 from 10.33.0.25/20 iif lo table viaeth0
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip rule add priority 1001 from 10.37.0.25/20 iif lo table viaeth1
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip rule add priority 1002 iif eth0 table viaeth1
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip rule add priority 1003 iif eth1 table viaeth0
Dec 24 02:35:37 multinic-phk3 policy-routing[1635]: + ip route flush cache
Dec 24 02:35:37 multinic-phk3 systemd[1]: Started Configure Policy Routing to behave as a virtual wire.
-- Subject: Unit policy-routing.service has finished start-up
-- Defined-By: systemd
-- Support: https://access.redhat.com/support
--
-- Unit policy-routing.service has finished starting up.
--
-- The start-up result is done.
```

**Comment by jeffmccune** [comment]
Released in [3.2.1](https://github.com/openinfrastructure/terraform-google-multinic/releases/tag/3.2.1)
